### ISSUE TITLE
Switch to JSON syntax highlighter for JSON text sections of commands

### ISSUE THREAD

**Issue by DrHenchman** [rubixninja314/vim-mcfunction#37]
First off, really nice plugin. Thanks for developing it!

I think it would be nice to switch to the JSON syntax highlighter for JSON text sections of commands. I managed to find an example of how to do this from the markdown syntax highlighter: https://github.com/plasticboy/vim-markdown/blob/master/ftplugin/markdown.vim#L810 however I am pretty lost when it comes to vim's plugin scripting language so I'm not complete sure how to translate this feature to this plugin, otherwise I would raise a PR myself.

**Comment by rubixninja314** [comment]
Thank you very much! What's kinda funny is I actually looked at that exact plugin to try and figure out a bit of vimscript, and I remember seeing that exact code and found it quite interesting that one could do that. And I totally get what you mean about vimscript being difficult, as I'm still starting to get the hang of it. And beyond that I did some really weird things on this that would likely be a bit difficult to figure out.

As for the JSON, that actually shouldn't be too hard, especially considering (ironically) said weird techniques that I recently added.

And lastly, if you don't mind me asking, how did you come across this plugin? It's the first project that I've put this much polish into and I'm really glad you like it!
